diff --git a/tidal.cabal b/tidal.cabal
index 7185e7b..da0019f 100644
--- a/tidal.cabal
+++ b/tidal.cabal
@@ -70,15 +70,15 @@ library
     , exceptions < 0.11
     , mtl >= 2.2
 
-executable tidal
-  hs-source-dirs: main
-  main-is: Main.hs
-  build-depends: base >=4.8 && < 5
-               , tidal
-               , hint < 0.10
-               , exceptions < 0.11
-               , async < 2.3
-  default-language:    Haskell2010
+-- executable tidal
+--  hs-source-dirs: main
+--  main-is: Main.hs
+--  build-depends: base >=4.8 && < 5
+--               , tidal
+--               , hint < 0.10
+--               , exceptions < 0.11
+--               , async < 2.3
+--  default-language:    Haskell2010
     
 test-suite tests
   type: exitcode-stdio-1.0
